src/NServiceBus.RabbitMQ.AcceptanceTests/When_using_a_custom_message_id_strategy.cs

```diff
@@ -0,0 +1,81 @@
+﻿namespace NServiceBus.RabbitMQ.AcceptanceTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using NServiceBus.AcceptanceTesting;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NServiceBus.Serialization;
+    using NServiceBus.Transports.RabbitMQ;
+    using NServiceBus.Unicast;
+    using NUnit.Framework;
+
+    public class When_using_a_custom_message_id_strategy
+    {
+        [Test]
+        public void Should_be_able_to_receive_messages_with_no_id()
+        {
+            var context = new Context();
+
+            Scenario.Define(context)
+                   .WithEndpoint<Receiver>(b => b.Given((bus, c) =>
+                   {
+                       var unicastBus = (UnicastBus)bus;
+                       var connectionManager = unicastBus.Builder.Build<IManageRabbitMqConnections>();
+
+                       var serializer = unicastBus.Builder.Build<IMessageSerializer>();
+
+                       using (var stream = new MemoryStream())
+                       {
+                           serializer.Serialize(new MyRequest(), stream);
+
+                           using (var channel = connectionManager.GetPublishConnection().CreateModel())
+                           {
+                               var properties = channel.CreateBasicProperties();
+
+                               //for now until we can patch the serializer to infer the type based on the root node
+                               properties.Headers = new Dictionary<string, object> { { Headers.EnclosedMessageTypes, typeof(MyRequest).FullName } };
```

Comment:
@SimonCropp @johnsimons @yvesgoeleven should we consider making the serializers smarter when it comes to inferring messages type to make external integrations easier? (json serializers also needs to the header to be set)

For XML we can match on the root node and for json we can use the $type thing?